cs-commons/src/main/java/io/cloudslang/content/utils/CollectionUtilities.java

```diff
@@ -0,0 +1,90 @@
+package io.cloudslang.content.utils;
+
+import io.cloudslang.content.constants.ExceptionsValues;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/** A Collection utility class that offers array, list and map conversions from a string.
+ * Created by victor on 01.09.2016.
+ */
+public final class CollectionUtilities {
+
+    private CollectionUtilities() {}
+
+    /** Splits the stringArray by the delimiter into an array of strings without ignoring the escaped delimiters
+     * @param stringArray the string to be split
+     * @param delimiter the delimiter by which to split the stringArray
+     * @return an array of Strings
+     */
+    @NotNull
+    public static String[] toArrayWithEscaped(@Nullable final String stringArray, @NotNull final String delimiter) {
+        if (StringUtils.isEmpty(stringArray)) {
+            return new String[0];
+        }
+        return StringUtils.splitByWholeSeparatorPreserveAllTokens(stringArray, delimiter);
+    }
+
+    /** Splits the stringArray by the delimiter into an array of strings ignoring the escaped delimiters
+     * @param stringArray the string to be split
+     * @param delimiter the delimiter by which to split the stringArray
+     * @return an array of Strings
+     */
+    @NotNull
+    public static String[] toArray(@Nullable final String stringArray, @NotNull final String delimiter) {
+        if (StringUtils.isEmpty(stringArray)) {
+            return new String[0];
+        }
+        final String regex = "(?<!\\\\)" + Pattern.quote(delimiter);
+        return stringArray.split(regex);
+    }
+
+    /** Splits the stringList by the delimiter into a List of strings without ignoring the escaped delimiters
+     * @param stringList the string to be split
+     * @param delimiter the delimiter by which to split the stringArray
+     * @return a list of Strings
+     */
+    @NotNull
+    public static List<String> toListWithEscaped(@Nullable final String stringList, @NotNull final String delimiter) {
+        return Arrays.asList(toArrayWithEscaped(stringList, delimiter));
```

Comment:
this is a list backed by the arra that cannot be changed y, so please use Lists.newArrayList to have a mutable list not backed by an array